firmware/EsaSpeedSwitch/EsaSpeedSwitch.ino

```diff
@@ -24,229 +24,238 @@
   };
-  
-  if(command == COMMAND_SPEED){
-    *(uint16_t *)&data[6] = (value * 252) / 10;
-  }
-  else 
-  {
+
+  if (arg == ARG_MAXSPEED)
+    *(uint16_t *)&data[6] = value * 252 / 10; 
+  else
     *(uint16_t *)&data[6] = value;
-  }
-    *(uint16_t *)&data[8] = calculateChecksum(data + 2);
 
-    RX_DISABLE;
-    for (int i = 0; i < iterations; i++)
-    {
-     Serial.write(data, sizeof(data) / sizeof(uint8_t));
-     delay(50);
-    }
-    RX_ENABLE;
+  *(uint16_t *)&data[8] = calculateChecksum(data + 2);
+
+  RX_DISABLE;
+  for (int i = 0; i < iterations; i++)
+  {
+    Serial.write(data, sizeof(data) / sizeof(uint8_t));
+    delay(50);
+  }
+  RX_ENABLE;
 }
 
 uint8_t readBlocking()
 {
-  while(!Serial.available()) delay(1);
+  while (!Serial.available())
+    delay(1);
   return Serial.read();
 }
 
-uint8_t receivePacket(uint8_t* throttle, uint8_t* brake)
+uint8_t receivePacket(uint8_t *throttle, uint8_t *brake, bool *dashButton)
 {
-  if(readBlocking() != 0x55)
+  if (readBlocking() != 0x55)
     return 0;
-  if(readBlocking() != 0xAA)
+
+  if (readBlocking() != 0xAA)
     return 0;
 
   uint8_t buff[256];
 
   uint8_t len = readBlocking();
   buff[0] = len;
-  if(len >= sizeof(buff) - 4)
+
+  if (len >= sizeof(buff) - 4)
     return 0;
 
   uint8_t addr = readBlocking();
   buff[1] = addr;
 
-  for(int i = 0; i < len; i++)
-  {
+  for (int i = 0; i < len; i++)
     buff[i + 2] = readBlocking();
-  }
 
   uint16_t actualChecksum = (uint16_t)readBlocking() | ((uint16_t)readBlocking() << 8);
   uint16_t expectedChecksum = calculateChecksum(buff);
-  if(actualChecksum != expectedChecksum)
+
+  if (actualChecksum != expectedChecksum)
     return 0;
 
   uint8_t success = 0;
-  
-  switch(addr)
+
+  switch (addr)
+  {
+  case 0x25:
   {
-    case 0x25:
+    switch (buff[2])
     {
-      switch (buff[2])
-      {
-        case 0x60:
-        {
-          *throttle = buff[5];
-          *brake = buff[6];
-          success = 1;
-        }
-        break;
-        case 0x64:
-        {
-          *throttle = buff[6];
-          *brake = buff[7];
-          success = 1;
-        }
-        break;
-      }
-    }
-    break;
-    case 0x27:
+    case 0x60:
     {
       *throttle = buff[5];
       *brake = buff[6];
       success = 1;
     }
     break;
+    case 0x64:
+    {
+      *throttle = buff[6];
+      *brake = buff[7];
+      success = 1;
+    }
+    break;
+    }
+  }
+  break;
+  case 0x27:
+  {
+    *throttle = buff[5];
+    *brake = buff[6];
+    success = 1;
+  }
+  break;
+  case 0x28:
+  {
+    *dashButton = (buff[10] == 0x01);
+    success = 1;
+  }
+  break;
   }
   return success;
 }
 
-void flashLED(int count, int duration)
+void flashLED(int count, int pause)
 {
   for (int i = 0; i < count; i++)
   {
     digitalWrite(LED_BUILTIN, HIGH);
-    delay(duration);
+    delay(pause);
     digitalWrite(LED_BUILTIN, LOW);
-    delay(100);
+
+    if (i < (count - 1))
+    {
+      delay(pause);
+    }
   }
 }
 
+void blinkEco()
```

Comment:
I like the idea of using the eco led as a feedback channel :)

Reply:
I also tried using the speedometer itself, but that didn't work... Yet 🤓